api/src/schema/index.js

```diff
@@ -12,6 +12,14 @@ const Schema = i18n => {
       eq: String
     }
 
+    type PaginatedResultSet {
+      hasNext: Boolean
+      hasPrevious: Boolean
+      next: String
+      previous: String
+      results: [Dwelling]
+    }
```

Comment:
Is this a convention or anything? 
There's a [JSON API spec](http://jsonapi.org/) that has some example structure like 

```json
{
  "links": {
    "self": "http://example.com/articles",
    "next": "http://example.com/articles?page[offset]=2",
    "last": "http://example.com/articles?page[offset]=10"
  },
  "data": [{
    "type": "articles",
    "id": "1",
    "attributes": {
      "title": "JSON API paints my bikeshed!"
    }
  }]
}
``` 

I am very much not in love with this suggested response structure but am wondering if you have any thoughts.

Reply:
In terms of convention, there is the [Relay Cusor Connections Spec](https://facebook.github.io/relay/graphql/connections.htm). This was created to provide a foundation for [Relay](https://facebook.github.io/relay/) to build automated pagination on top of. Since we are not using Relay (and only a few people are in general, and most of the momentum seems to be around Apollo at the moment) implementing the Relay spec for this endpoint feels like overkill.

I ended up trying to find the simplest cursor pagination I could (something plausibly usable directly by an end user), found [this library](https://github.com/mixmaxhq/mongo-cursor-pagination) and adopted it's [return type](https://github.com/mixmaxhq/mongo-cursor-pagination#find).

Apollo seems to have it's own [thoughts](https://dev-blog.apollodata.com/tutorial-pagination-d1c3b3ee2823) about [pagination](https://dev-blog.apollodata.com/understanding-pagination-rest-graphql-and-relay-b10f835549e7)(https://www.apollographql.com/docs/react/recipes/pagination.html#cursor-pages), which might provide some nicer porcelain over the same plumbing. Let's revisit this.

Reply:
Cool cool, just checking. Iterate wildly and all that.